keras_core/layers/convolutional/conv_transpose_test.py

```diff
@@ -749,3 +750,90 @@ def test_conv3d_transpose(
             dilation_rate,
         )
         self.assertAllClose(outputs, expected, atol=1e-5)
+
+    @parameterized.product(
+        kernel_size=list(range(1, 5)),
+        strides=list(range(1, 5)),
+        padding=["same", "valid"],
+        output_padding=[None] + list(range(1, 5)),
+    )
+    def test_conv1d_vs_tf_implementation(
+        self, kernel_size, strides, padding, output_padding
+    ):
+        """Compares results from keras-core implementation to the results from
+        tf-keras, for a 1D input shape of length 3.
+        """
+
+        # output_padding cannot be greater than strides
+        if isinstance(output_padding, int) and output_padding >= strides:
+            pytest.skip(
+                "`output_padding` greater than `strides` is not supported"
+            )
+
+        input = np.ones(shape=(1, 3, 1))
+        kernel_weights = np.arange(1, kernel_size + 1).reshape(
+            (kernel_size, 1, 1)
+        )
+
+        # TF layer
+        tf_layer = tf.keras.layers.Conv1DTranspose(
```

Comment:
OK, I now used `np_conv1d_transpose` to remove the dependency on tf.keras.
Note that it's not 100% fool-proof, because `np_conv1d_transpose` depends on some internal keras-core function (for instance to get the output shape), so if those are wrong, the test could still pass.